### ISSUE TITLE
How to  add a user with the IDP details to the realm using the API

### ISSUE THREAD

**Issue by althaf004** [marcospereirampj/python-keycloak#609]
I needed to create a user to the realm with its IDP details.

is it some parameters i needed to have along with https://github.com/marcospereirampj/python-keycloak/blob/7cfad72a68346ca4cbcba69f9e8808091ae47daa/src/keycloak/keycloak_admin.py#L551 API?

```
      import json
      from keycloak import KeycloakOpenID
      from keycloak import KeycloakAdmin
      try:         
        keycloak_admin = KeycloakAdmin(server_url=api_base_url,
                                          username=admin_username,
                                          password=admin_password,
                                          realm_name=master_realm,                                     
                                          verify=True)    
        keycloak_admin.realm_name = user_realm_name                                
        new_user = keycloak_admin.create_user({"email":email,
                              "username": username,
                              "enabled": True,
                              "firstName":firstname,
                              "lastName": lastname,
                              "credentials": [{"value": value_password,"type":  "password",}]})
```

Thanks
Althaf

**Comment by carbonleakage** [comment]
If I understand correctly you have an admin user in the master realm, you would like to create a user in another realm? If so [this should work](https://github.com/marcospereirampj/python-keycloak/issues/610#issuecomment-2439395181). 

When initiating the `KeycloakAdmin` set the `user_realm_name= 'master'` or the appropriate realm where the admin user is defined, and set the `realm_name` to the realm where you want new user to be created.

**Comment by althaf004** [comment]
@carbonleakage

You’re correct that adding a user from the master realm to another realm is the case here. However, along with the user details, I also need to add the IDP (Identity Provider) details of the user.

This is where I’m stuck.

**Comment by ryshoooo** [comment]
Hi @althaf004 

What specific IdP details do you wish to add?

Usually, data from external IdPs are propagated to keycloak users via IdP mappers, where you can decide the logic of where the data should be stored or translated into. Usually I store those as user attributes, but you can do more, f.e. translate specific values into role assignments etc. So what do you wish add?
